app/services/letter_sending_service.rb

```diff
@@ -46,6 +50,14 @@ def deliver!
     neighbour.touch :last_letter_sent_at
   end
 
+  def deliver_batch!(neighbours)
+    @batch = neighbours.first.consultation.neighbour_letter_batches.new(text: letter_content)
+
+    neighbours.each do |neighbour|
+      deliver!(neighbour)
+    end
+  end
```

Comment:
Can we add a rescue block here to continue the batch processing even if a neighbour fails and send this error to AppSignal

```ruby
rescue => e
    # Log and continue processing the rest of the batch
    Appsignal.send_error(e)
    next
  end
```

Reply:
Might be `report_error` now actually